llama_stack/providers/impls/meta_reference/inference/parallel_utils.py

```diff
@@ -23,11 +22,65 @@
     get_model_parallel_src_rank,
 )
 
+from llama_models.llama3.api.datatypes import Message, ToolPromptFormat
+
+from pydantic import BaseModel
+
 from torch.distributed.launcher.api import elastic_launch, LaunchConfig
 
+from .generation import TokenResult
+
+
+class InferenceArgs(BaseModel):
+    messages: List[Message]
+    temperature: float
+    top_p: float
+    max_gen_len: int
+    logprobs: bool
+    tool_prompt_format: ToolPromptFormat
+
+    # avoid converting llama_models types to BaseModel for now
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class ReadyRequest(BaseModel):
+    message: str = "READY?"
+
+
+class ReadyResponse(BaseModel):
+    message: str = "YES READY"
+
+
+class EndSentinel(BaseModel):
+    message: str = "__end_sentinel__"
+
+
+class CancelSentinel(BaseModel):
+    message: str = "__cancel_sentinel__"
 
-_END_SENTINEL = "__end_sentinel__"
-_CANCEL_SENTINEL = "__cancel_sentinel__"
+
+class TaskRequest(BaseModel):
+    task: InferenceArgs
+
+
+class TaskResponse(BaseModel):
+    result: TokenResult
+
+
+class ExceptionResponse(BaseModel):
+    error: str
+
+
+Message = Union[
```

Comment:
uh oh are you redefining `Message` from `llama_models`?

Reply:
Great catch!